src/test/java/core/basesyntax/service/OperationServiceTest.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.impl.ParseServiceImpl;
+import core.basesyntax.service.impl.ReadServiceImpl;
+import core.basesyntax.service.impl.ReportServiceImpl;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.impl.BalanceOperationHandler;
+import core.basesyntax.strategy.impl.PurchaseOperationHandler;
+import core.basesyntax.strategy.impl.ReturnOperationHandler;
+import core.basesyntax.strategy.impl.SupplyOperationHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class OperationServiceTest {
+    private static final String INPUT_PATH = "src/test/resources/input.csv";
+    private static OperationStrategy operationStrategy;
+    private static OperationService operationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(
+                FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler());
+        operationHandlerMap.put(
+                FruitTransaction.Operation.RETURN, new ReturnOperationHandler());
+        operationHandlerMap.put(
+                FruitTransaction.Operation.PURCHASE, new PurchaseOperationHandler());
+        operationHandlerMap.put(
+                FruitTransaction.Operation.BALANCE, new BalanceOperationHandler());
+        operationStrategy = new OperationStrategy(operationHandlerMap);
+    }
+
+    @AfterAll
+    static void afterAll() {
+        Storage.STORAGE.clear();
+    }
+
+    @Test
+    void validResult_Ok() {
+        String expected = "fruit,quantity\n"
+                + "banana,152\r\n"
+                + "apple,90\r\n";
+        List<String> inputData = new ReadServiceImpl().readInputData(INPUT_PATH);
+        List<FruitTransaction> parseData = new ParseServiceImpl().parseInputData(inputData);
+        operationService = new OperationService(operationStrategy);
+        operationService.processOperation(parseData);
+        String actual = new ReportServiceImpl().createReport();
```

Comment:
[Unit testing is isolated testing](https://mate-academy.github.io/jv-program-common-mistakes/java-core/solid/fruit-shop-tests#unit-testing-is-isolated-testing). [Ensure that you test your services and their independently](https://mate-academy.github.io/jv-program-common-mistakes/java-core/solid/fruit-shop-tests#ensure-that-you-test-your-services-and-their-independently). I think you are using a lot of classes here